modules/core/shared/src/main/scala/lucuma/core/model/sequence/ABBA.scala

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2016-2025 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.core.model.sequence
+
+import cats.Eq
+import cats.derived.*
+import cats.syntax.eq.*
+import lucuma.core.math.Offset
+import monocle.Focus
+import monocle.Lens
+
+/**
+ * An ABBA pattern containing exactly 4 offsets with symmetric structure.
+ */
+case class ABBA(a: Offset, b: Offset) derives Eq:
```

Comment:
I think this is usually right but sometimes the two a’s and/or b’s aren’t exactly the same. We have to leave open the possibility that they differ or it won’t be usable by F2 long slit at least. Maybe a constructor that takes an ‘a’ and a ‘b’ but in general I think we need four positions.

Reply:
Got his from science:
```
sw
[5:20] are they always symmetric?
[5:21]-a, +a, +a, -a

science:
[5:21] yes, e.g. +3, -2, -2, +3

sw
[5:26] no odd astronomer will do -1, 3, -4, 5?

science:
[5:27 PM] they may, but that's not ABBA :slightly_smiling_face:
```

Reply:
From the requirements spreadsheet examples:

<img width="819" alt="Screenshot 2025-07-04 at 10 06 06 AM" src="https://github.com/user-attachments/assets/ffece98d-a98c-4296-a4b0-8bb513b72a12" />

The code looks great though, and it will probably be used in other modes so :+1:

Reply:
No worries, maybe we eventually could use an `Abba-ish` class with 4 positions but for now F2 Long Slit can continue working just fine with a `List` + validation.